modules/custom/social_language/modules/social_content_translation/social_content_translation.install

```diff
@@ -5,75 +5,121 @@
-      $definition->getKey('content_translation_source'),
-      $definition->getKey('content_translation_outdated'),
-      $definition->getKey('revision_translation_affected'),
-    ];
+/**
+ * Install translation configuration.
+ *
+ *  "Content Translation" module provide UI where we can enabled/disable
+ *  translations for entity types we want. So here we enabling translations
+ *  for modules if it have "language.content_settings*" configuration in
+ *  "config/optional" folder.
+ */
+function social_content_translation_install_language_settings_for_entities() {
+  $config_storage = \Drupal::service('config.storage');
+
+  // Get enabled modules.
+  $modules = \Drupal::moduleHandler()->getModuleList();
+  foreach ($modules as $module_name => $extension) {
+    // Check if modules have target configuration.
+    $config_path = drupal_get_path('module', $module_name) . '/config/optional';
+    $source = new FileStorage($config_path);
+    if (!($list = $source->listAll('language.content_settings'))) {
+      // Do nothing.
+      continue;
+    }
 
-    foreach (array_keys($bundle_info) as $bundle) {
-      $fields = $field_manager->getFieldDefinitions($entity_type_id, $bundle);
-      foreach ($fields as $field) {
-        $field_config = $field->getConfig($bundle);
-        // Only disable translatable fields and always allow some exceptions
-        // that are required for content translation to function properly.
-        if ($field_config->isTranslatable() && !in_array($field->getName(), $exceptions)) {
-          $field_config
-            ->setTranslatable(FALSE)
-            ->save();
-        }
+    foreach ($list as $config_name) {
+      // Probably, configuration can be already existed if module provided
+      // it was enabled after "Content Translation" module was enabled.
+      // Otherwise lets install it.
+      if (!$config_storage->exists($config_name)) {
+        $file = $source->read($config_name);
+        // We create a config as an entity because it allows to update field
+        // definitions for translated fields.
+        // @see content_translation_language_content_settings_insert()
+        $config = ContentLanguageSettings::create($file);
+        $config->save();
       }
     }
   }
 
-  // Ensure entity and menu router information are correctly rebuilt.
-  \Drupal::service('entity_type.manager')->clearCachedDefinitions();
+  \Drupal::service('entity_type.bundle.info')->clearCachedBundles();
   \Drupal::service('router.builder')->setRebuildNeeded();
 }
 
+/**
+ * Fill in translations fields for entity after enabling translation.
+ *
+ *  If we have entities before translation was enabled for them after
+ *  enabling and updating fields definitions we got empty translations
+ *  fields (like "content_translation_source", "content_translation_uid" etc.
+ *  The functions tries to fix this and if possible fill these fields.
+ */
+function social_content_translation_fill_in_translations_fields($entity_type) {
```

Comment:
@nechai If you don't use this function in Open Social profile I think it's better to remove it.